### ISSUE TITLE
Question about PPO2 learning rate and optimizer

### ISSUE THREAD

**Issue by greg2paris** [hill-a/stable-baselines#374]
Hello,
I am training a ppo2 model, but their are some details that aren't very clear for me.
When I create the ppo2 model for the first time, I use a learning rate of 0.001
When I check the learning rate with :
print("Learning Rate: ", model.learning_rate)
It shows exactly that, but when it run for 1 full epoch, then the : 
print("Learning Rate: ", model.learning_rate)
shows : 
"Learning Rate:  <function constfn.<locals>.func at 0x7f163c5e7158>"

So I guess the learning rate is used by an optimizer that change the learning rate automaticaly like adam.
**Is their a way to show the current learning_rate from that function?**

**Can the learning rate increase with time, or it can only decrease ?**

Because when I load a trained model from a file, its like the training is starting from the start again.

**So my question is, do the learning_rate of the imported model is the same as it was when the model was saved, or do the learning_rate is reset every time the model is loaded?**

**Last question, how can we use a different optimizer for the ppo2**

Thank you for your reading, hope someone can help me with this.
Greg

**Comment by greg2paris** [comment]
> Hello,
> I am training a ppo2 model, but their are some details that aren't very clear for me.
> When I create the ppo2 model for the first time, I use a learning rate of 0.001
> When I check the learning rate with :
> print("Learning Rate: ", model.learning_rate)
> It shows exactly that, but when it run for 1 full epoch, then the :
> print("Learning Rate: ", model.learning_rate)
> shows :
> "Learning Rate: <function constfn..func at 0x7f163c5e7158>"
> 
> So I guess the learning rate is used by an optimizer that change the learning rate automaticaly like adam.
> **Is their a way to show the current learning_rate from that function?**
> 
> **Can the learning rate increase with time, or it can only decrease ?**
> 
> Because when I load a trained model from a file, its like the training is starting from the start again.
> 
> **So my question is, do the learning_rate of the imported model is the same as it was when the model was saved, or do the learning_rate is reset every time the model is loaded?**
> 
> **Last question, how can we use a different optimizer for the ppo2**
> 
> Thank you for your reading, hope someone can help me with this.
> Greg

ps: I tried to put a label with "question" for the last 15min but I realy didn't find how to do, sorry for that.

**Comment by araffin** [comment]
Hello,

> Is their a way to show the current learning_rate from that function?

Answer is in the [code](https://github.com/hill-a/stable-baselines/blob/master/stable_baselines/ppo2/ppo2.py#L330) or in the [sac documenation](https://stable-baselines.readthedocs.io/en/master/modules/sac.html#parameters) (which uses the same type of callable):
"it can be a function of the current progress (from 1 to 0)"

so you have to call it to now the current learning rate:
```python
current_progress = 1.0  # from 1 (start of training) to 0
print(model.learning_rate(1))  # initial learning rate
```

>Can the learning rate increase with time, or it can only decrease ?

It is a callable, it can be anything.

>do the learning_rate is reset every time the model is loaded?

It is reset for now (see #301 )

>Last question, how can we use a different optimizer for the ppo2

You need to have a custom version of stable baselines.

**Comment by greg2paris** [comment]
ok thanks !
